storage/storage-interface/src/state_view.rs

```diff
@@ -62,3 +62,24 @@ impl LatestDbStateView for Arc<dyn DbReader> {
         })
     }
 }
+
+pub trait DbStateViewAtVersion {
```

Comment:
why we need this?

Reply:
When we do multi-hop queries (like indirection to a table item), we want to base the queries on a consistent db view.

The current implementation under /account is wrong -- although it doesn't do any indirection, it always interprets resources at a certain version with the LATEST modules. (Before the latest `latest_state_view()` thing was introduced, it was worse -- the converter fetches different modules at changing versions.)